File: packages/types/src/task.ts
export type TaskType = 'pick' | 'putaway' | 'replenish' | 'count' | 'receive';
export type TaskStatus = 'unassigned' | 'assigned' | 'in_progress' | 'complete' | 'exception';
export type TaskLineStatus = 'open' | 'complete' | 'short' | 'skipped';
export type OrderStatus = 'new' | 'released' | 'picking' | 'packed' | 'shipped';
export type OrderLineStatus = 'open' | 'partial' | 'complete' | 'short';
export type WaveStatus = 'planning' | 'released' | 'active' | 'complete';

export interface TaskLineExpanded {
  id: string;
  task_id: string;
  order_line_id: string;
  item: {
    sku: string;
    name: string;
    uom: string;
  };
  location: {
    barcode: string;
    zone: string;
    aisle: string | null;
    bay: string | null;
    level: string | null;
  };
  lot_number: string | null;
  qty_required: number;
  qty_picked: number;
  status: TaskLineStatus;
}

export interface TaskLine {
  id: string;
  task_id: string;
  order_line_id: string;
  item_id: string;
  location_id: string;
  lot_number: string | null;
  qty_required: number;
  qty_picked: number;
  status: TaskLineStatus;
  picked_by: string | null;
  picked_at: Date | null;
}

export interface Task {
  id: string;
  tenant_id: string;
  wave_id: string | null;
  type: TaskType;
  status: TaskStatus;
  priority: number;
  assigned_to: string | null;
  assigned_at: Date | null;
  completed_at: Date | null;
  source_location_id: string | null;
  dest_location_id: string | null;
  created_at: Date;
}

export interface Wave {
  id: string;
  tenant_id: string;
  status: WaveStatus;
  released_at: Date | null;
  released_by: string | null;
}

File: android/app/src/main/java/com/kinetic/wms/data/model/Task.kt
package com.kinetic.wms.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
enum class TaskStatus {
    @SerialName("unassigned") UNASSIGNED,
    @SerialName("assigned") ASSIGNED,
    @SerialName("in_progress") IN_PROGRESS,
    @SerialName("complete") COMPLETE,
    @SerialName("exception") EXCEPTION,
}

@Serializable
enum class TaskLineStatus {
    @SerialName("open") OPEN,
    @SerialName("complete") COMPLETE,
    @SerialName("short") SHORT,
    @SerialName("skipped") SKIPPED,
}

@Serializable
data class TaskLineItem(
    val sku: String,
    val name: String,
    val uom: String,
)

@Serializable
data class TaskLineLocation(
    val barcode: String,
    val zone: String,
    val aisle: String? = null,
    val bay: String? = null,
    val level: String? = null,
)

@Serializable
data class TaskLineExpanded(
    val id: String,
    @SerialName("qty_required") val qtyRequired: Int,
    val item: TaskLineItem,
    val location: TaskLineLocation,
    val status: TaskLineStatus = TaskLineStatus.OPEN,
)

@Serializable
data class Task(
    val id: String,
    @SerialName("assigned_to") val assignedTo: String? = null,
    val status: TaskStatus,
    val priority: Int = 0,
    @SerialName("task_lines") val taskLines: List<TaskLineExpanded> = emptyList(),
)

@Serializable
data class TaskLineCompleteResponse(
    val id: String,
    @SerialName("qty_picked") val qtyPicked: Int,
    val status: TaskLineStatus,
    @SerialName("picked_by") val pickedBy: String? = null,
    @SerialName("picked_at") val pickedAt: String? = null,
)

@Serializable
data class AdvanceLineResponse(
    @SerialName("has_more") val hasMore: Boolean,
    @SerialName("current_line_index") val currentLineIndex: Int,
    @SerialName("next_task_line") val nextTaskLine: TaskLineExpanded? = null,
)

@Serializable
data class CompleteTaskLineRequest(
    @SerialName("qty_picked") val qtyPicked: Int,
    @SerialName("lot_number") val lotNumber: String? = null,
)

@Serializable
data class TaskCompleteResponse(
    val id: String,
    val status: TaskStatus,
    @SerialName("completed_at") val completedAt: String? = null,
)

@Serializable
data class EscalateRequest(
    @SerialName("worker_id") val workerId: String,
    @SerialName("step_id") val stepId: String,
)

@Serializable
data class EscalateResponse(
    val acknowledged: Boolean,
    @SerialName("task_id") val taskId: String,
)
